xbmc/filesystem/CurlFile.cpp

```diff
@@ -925,6 +940,16 @@ void CCurlFile::Reset()
   m_state->m_cancelled = false;
 }
 
+void CCurlFile::SetProxy(ProxyType type, const std::string &host,
```

Comment:
if you make type a string, you don't have the need to expose internal definitions like ProxyType